### PR TITLE
feat: port media/base

### PR DESCRIPTION
**Pull request by joeyparrish**
This removes all chromium dependencies from media/base/ and completes the build system in CMake.

The ClosureThread class and its classes were removed, as they were specific to chromium base.  ClosureThread has been replaced by std::thread.

The byte-swapping utilities in network_util.cc have been removed and replaced with absl.

generate_unique_temp_path() was split out of file_unittest.cc into file_test_util.cc, where other test suites could make use of it.

WARN_UNUSED_RESULT was replaced with the C++ standard attribute [[nodiscard]].

The base::Clock interface was replaced with a typedef for a function pointer that returns the current time.

This re-enables the tests in http_key_fetcher_unittest.cc by using httpbin.org.

Issue #1047 (CMake porting)
Issue #346 (absl porting)

### CODE REVIEW

File: packager/media/base/container_names.cc [hunk 1]
```diff
@@ -1726,41 +1727,44 @@ MediaContainerName DetermineContainer(const uint8_t* buffer, int buffer_size) {
 
 MediaContainerName DetermineContainerFromFormatName(
     const std::string& format_name) {
-  if (base::EqualsCaseInsensitiveASCII(format_name, "aac") ||
-      base::EqualsCaseInsensitiveASCII(format_name, "adts")) {
+  std::string normalized_format_name = format_name;
+  std::transform(format_name.begin(), format_name.end(),
+                 normalized_format_name.begin(), ::tolower);
+
+  if (normalized_format_name == "aac" || normalized_format_name == "adts") {
```
Comment (theodab) [inline packager/media/base/container_names.cc:1734]: Now that this is no longer using a custom comparison method, couldn't it now be a `switch` statement?

**Review by theodab — CHANGES_REQUESTED** [review]

**Comment by joeyparrish** [comment]
This includes the changes in PR #1132 and #1133.  With that, builds are (finally) all passing.

**Review by theodab — APPROVED** [review]

File: packager/media/base/macros.h [hunk 2]
```diff
@@ -46,9 +46,7 @@
 #endif
 
 #ifndef FALLTHROUGH_INTENDED
-#define FALLTHROUGH_INTENDED \
-  do {                       \
-  } while (0)
+#define FALLTHROUGH_INTENDED
```
Comment (kqyang) [inline packager/media/base/macros.h:49]: What is the reason of removing the original definition?

File: packager/media/base/playready_pssh_generator.cc [hunk 3]
```diff
@@ -59,6 +57,45 @@ std::vector<uint8_t> ConvertGuidEndianness(const std::vector<uint8_t>& input) {
...
+  if (location != std::string::npos) {
+    str->replace(location, from.size(), to);
+  }
+}
+
+void AesEcbEncrypt(const std::vector<uint8_t>& key,
+                   const std::vector<uint8_t>& plaintext,
+                   std::vector<uint8_t>* ciphertext) {
+  CHECK_EQ(plaintext.size() % AES_BLOCK_SIZE, 0u);
+  ciphertext->resize(plaintext.size());
```
Comment (kqyang) [inline packager/media/base/playready_pssh_generator.cc:73]: So for ECB, the output size does not need to be one block more than the input size?

**Review by kqyang — APPROVED** [review]
lgtm

File: packager/media/base/macros.h (hunk 2 above)
Comment (joeyparrish) [reply packager/media/base/macros.h:49]: In an earlier PR, you advised me to remove the do/while(0) around some other macro definition.  So I did the same here.  I don't recall it causing any issues.

File: packager/media/base/playready_pssh_generator.cc (hunk 3 above)
Comment (joeyparrish) [reply packager/media/base/playready_pssh_generator.cc:73]: I double-checked the tests with valgrind, and got no errors.  Then I noticed that this code is not actually covered by tests...

I'll add an extra block.  :-(

I think we will need to do some code coverage analysis at some point.  I'll file an issue for that, too.

**Pull request merged**